addons/binding/org.openhab.binding.cups/src/main/java/org/openhab/binding/cups/internal/discovery/PrinterDiscoveryParticipant.java

```diff
@@ -0,0 +1,99 @@
+/**
+* Copyright (c) 2015 openHAB UG (haftungsbeschraenkt) and others.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Eclipse Public License v1.0
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v10.html
+*/
+package org.openhab.binding.cups.internal.discovery;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import javax.jmdns.ServiceInfo;
+
+import org.cups4j.CupsClient;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.eclipse.smarthome.io.transport.mdns.discovery.MDNSDiscoveryParticipant;
+import org.openhab.binding.cups.CupsBindingConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * discovers cups servers announced by mDNS
+ * @author Tobias Bräutigam
+ *
+ */
+public class PrinterDiscoveryParticipant implements MDNSDiscoveryParticipant {
```

Comment:
Shouldn't this be rather a CupsServerDiscoveryParticipant instead of a PrinterDiscoveryParticipant?